app/components/SpeakEasyForm.js

```diff
@@ -2,14 +2,18 @@ import React from "react";
 import CSSModules from "react-css-modules";
 import styles from "../styles/components/speak-easy-form.css";
 import { reduxForm } from "redux-form";
-import { postSpeakEasy } from "../actions/speakEasy";
+import { postSpeakEasy, updateSpeakEasy } from "../actions/speakEasy";
 import { getValues } from 'redux-form';
 
 class SpeakEasyForm extends React.Component {
   handleSubmission(e) {
     e.preventDefault();
-    const { dispatch, values } = this.props;
-    dispatch(postSpeakEasy(values));
+    const { dispatch, values, fields: { name }, id } = this.props;
+    if (name.defaultValue) {
```

Comment:
Actually would be way nice to pass the context of whether this form is new or update down rather than litter it with the same checks